fix (mobile): Fix slow album thumbnail generation for album picker

a-elkaim opened the pull request:
When generating the thumbnail of an album, all of the pictures of the album are retrieved but only the first picture of the album is used.

Retrieving all of the pictures of the album at once can cause huge performance issues on large albums.

Since only the first picture is used to generate the thumbnail, this commit uses the getAssetListPaged method instead of getAssetListRange, effectively retrieving only the first picture of the album.

Note that this issue has been observed on Android but may also happen on iOS devices.

Commit:
[BUGFIX] Fix slow album thumbnail generation

When generating the thumbnail of an album, all of the pictures of the
album are retrieved but only the first picture of the album is used.

Retrieving all of the pictures of the album at once can cause huge performance
issues on large albums.

Since only the first picture is used to generate the thumbnail, this commit
uses the getAssetListPaged method instead of getAssetListRange, effectively
retrieving only the first picture of the album.

EnochPrime commented on mobile/lib/modules/backup/providers/backup.provider.dart:
```diff
@@ -217,8 +217,7 @@ class BackupNotifier extends StateNotifier<BackUpState> {
-            await album.getAssetListRange(start: 0, end: assetCountInAlbum);
+        final assetList = await album.getAssetListPaged(page: 0, size: 1);
 
         if (assetList.isNotEmpty) {
```
If `assetList` could potentially be empty could the wrapping `assetCount > 0` be removed? Then it also shouldn't be necessary to query the asset count at all here.

azsde replied:
I don't think so as we do not want to list empty albums, that being said we can probably remove `if (assetList.isNotEmpty)`

What do you think @alextran1502 ?

EnochPrime replied:
Honestly I haven't looked at the implementation of each, but it seems logical to me that an asset count greater than 0 would be synonymous with an non empty list of assets from 0 to asset count.

I don't think an album with 0 assets would be able to return a non zero amount of assets.

azsde replied:
Ok, i've uploaded another commit

Commit:
[DEVMINOR] Remove unecessary check

As we already check for the number of assets in the album, when
fetching assets using `album.getAssetListPaged` the returned result
won't be empty.
